src/Features/VisualBasic/Portable/ConvertLinq/VisualBasicConvertLinqQueryToLinqMethodProvider.vb

```diff
@@ -0,0 +1,234 @@
+                        Return Nothing ' Too complicated for query method syntax.
+                    Case SyntaxKind.AggregateClause
+                        Return Nothing ' Too complicated for query method syntax.
+                    Case SyntaxKind.GroupByClause
+                        Return Nothing ' Group by in query and in method are not 100% equivalent. Consdier support of some group by in the future.
+                    Case Else
+                        Return Nothing
+                End Select
+            End Function
+
+            Private Function ProcessFromClause(parentExpression As ExpressionSyntax, fromClause As FromClauseSyntax) As ExpressionSyntax
+                Dim expression = parentExpression
+                Dim variables = fromClause.Variables
+                If variables.Count > 1 Then
+                    Return Nothing ' Do not support more than 1 variable. It complicates query methods.
+                End If
+
+                Dim variable = variables.First()
+                If parentExpression Is Nothing Then
+                    expression = variable.Expression
+
+                    If variable.AsClause IsNot Nothing Then
+                        Dim identifier = SyntaxFactory.IdentifierName(NameOf(Enumerable.Cast))
+                        Dim generic = SyntaxFactory.GenericName(
+                            identifier.Identifier,
+                            SyntaxFactory.TypeArgumentList(SyntaxFactory.SingletonSeparatedList(variable.AsClause.Type)))
+                        expression = SyntaxFactory.InvocationExpression(
+                            SyntaxFactory.MemberAccessExpression(
+                            SyntaxKind.SimpleMemberAccessExpression,
+                            expression,
+                            SyntaxFactory.Token(SyntaxKind.DotToken), generic),
+                            SyntaxFactory.ArgumentList())
+                    End If
+
+                    Return expression
+                End If
+
+                Dim anonymousFunctions = FindAnonymousFunctionsFromParentInvocationOperation(variable.Expression)
+                Dim firstLambda = CreateLambdaExpression(anonymousFunctions.First())
+                Dim secondArgumentAnonymousFunction = anonymousFunctions.Last()
+                Dim returnedValue = DirectCast(secondArgumentAnonymousFunction.Body.Operations.First(), IReturnOperation).ReturnedValue
+                Dim secondLambda As LambdaExpressionSyntax
+                If returnedValue.Kind = OperationKind.AnonymousObjectCreation Then
+                    Dim objectCreation = DirectCast(returnedValue, IAnonymousObjectCreationOperation)
+                    Dim fieldInitializers = objectCreation.Initializers.
+                        Select(Function(initializer) SyntaxFactory.InferredFieldInitializer(
+                        SyntaxFactory.IdentifierName(BeautifyName(DirectCast(initializer, IParameterReferenceOperation).Parameter.Name))))
```

Comment:
>SyntaxFactory.IdentifierName(BeautifyName(DirectCast(initializer, IParameterReferenceOperation).Parameter.Name)))) [](start = 24, length = 114)

I don't think this is going to work for a query that consists of three Froms in a row. Also, I believe the following Let could be merged into this selector. In general, I think we should be serializing IOperation tree to syntax here (rather than guessing what do we have) and be prepared to deal with user code inside, with user provided names, so InferredFieldInitializer is probably a bad idea.